Physical Location report enhancement, refs #12255

sbreker opened the pull request:
This commit corrects a couple issues with the physical location report:

- When physical object records had null values in the type_id field,
or other fields, the report would crash. This should no longer occur.
- Fields for which there was no current translation would be blank.
This should no longer occur.
- Characters with diacritics that occurred in the report title would
be garbled. This has been corrected.
- The box label report was having similar display issues of characters
with diacritics - this should also be fixed.

sbreker committed:
Physical Location report enhancement, refs #12255

This commit corrects a couple issues with the physical location report:

- When physical object records had null values in the type_id field,
or other fields, the report would crash. This should no longer occur.
- Fields for which there was no current translation would be blank.
This should no longer occur.
- Characters with diacritics that occurred in the report title would
be garbled. This has been corrected.
- The box label report was having similar display issues of characters
with diacritics - this should also be fixed.

jraddaoui reviewed (commented):
Hi @sbreker,

Is the error happening in the render methods when `null` is being sent? If so, I think it should be fixed in there as it may happen in other places.

jraddaoui commented on lib/job/arGenerateReportJob.class.php:
```diff
@@ -155,7 +155,19 @@ private function getStorageLocationsResults()
+    $storageLocations = QubitPhysicalObject::get($criteria);
+    $results = array();
+
+    foreach($storageLocations as $item)
```
I think some logic in the template in this case would be better, to avoid looping twice over the results. Not sure if this could be fixed in the render methods plus getting the values with source culture fallback in the templates.

sbreker committed:
Remove irrelevant fix, refs #12255

The fix to not crash on null values is specific to 2.4. This fix has been
removed from this version. Issues still exist to do with i18n and utf-8
encoding. These fixes have been retained.

sbreker commented:
I have updated the PR with a second commit. I have removed the fix for dealing with null values - this issue only exists in 2.4.x. Fixes for i18n and utf-8 are still present. 

It will be simplest to review using the "Files Changed" tab so you can view the net change.

sbreker commented:
@jraddaoui ^^.  No rush!

jraddaoui reviewed (approved):
Thanks @sbreker!
